lib/external_service_new/zendesk_downloader.rb

```diff
@@ -0,0 +1,51 @@
+module ExternalServiceNew
+  class ZendeskDownloader
+    def initialize(dispatcher:, logger:nil)
+      @dispatcher = dispatcher
+      @logger     = logger
+    end
+
+    def each_list(&block)
+      return to_enum(:each_list) unless block
+      [[
+        dispatcher.site_url,
+        "#{dispatcher.site_url}'s customers"
+      ]].each(&block)
```

Comment:
right now, you still seem to have 1 iteration on the first element of the array, wich is  [dispatcher_url, "[XXX]'s customers"]

Same question as before, I'm not sure i understant why you are forcing #each_list to get one list only.  Is it because Zendesk does not return multiple mailing lists but instead one huge list containing everyone (with filter options)?